esmtools/prediction.py

```diff
@@ -609,9 +604,78 @@ def PM_compute(ds, control, metric=_pearson_r, comparison=_m2m, anomaly=False,
         raise ValueError('specify metric argument')
 
 
+def DP_compute_skill(dp, reference, nlags=None):
+    """
+    Compute a predictability skill score for a reconstruction framework dataset.
+    
+    This is currently set up to just run a simple correlation (i.e., ACC) but
+    other metrics should be implemented if needed in the future (as in PM_compute)
+
+    Note that if reference is the reconstruction, the output correlation coefficients
+    are for potential predictability. If the reference is observations, the ouput
+    correlation coefficients are actual skill.
+
+    Parameters
+    ----------
+    dp : xarray object
+        Expected to follow package conventions (and should be an ensemble mean)
+        `ensemble` : dim of initialization dates
+        `time` : dim of lead years from those initializations
+        Additional dims can be lat, lon, depth, etc. but should not be individual
+        members.
+    reference : xarray object
+        Reconstruction or observations over same time period
+    nlags : int (default length of `time` dim)
+        How many lags to compute skill/potential predictability out to
+
+    Returns
+    -------
+    skill : xarray object
+        Predictability with main dimension `lag`
+    """
+    _check_xarray(dp)
+    _check_xarray(reference)
+    if 'member' in _get_dims(dp):
+        print("Taking ensemble mean...")
+        dp = dp.mean('member')
+    N = dp['ensemble'].size
+    if nlags is None:
+        nlags = dp['time'].size
+    skill = xr.concat([xr_corr(dp.isel(time=i), reference, lag=i,
+                dim='ensemble') for i in range(0, nlags)], 'lag')
```

Comment:
It depends on what we would like to see on the x-axis of plots. 
I would give it the name 'time' as in the lead time before, but its a matter of taste.

Reply:
I switched to 'lead time'